### DISCUSSION TITLE
Drawers manager (like modals manager)

### DISCUSSION THREAD

**Discussion by fengye87** [mantinedev/mantine#5869] [Feature requests]
Modals manager makes it very easy to open any dynamic modal whenever needed using the `modals.open()` method. But when it comes to Drawer, things becomes not so easy. Since Drawer and Modal share some common usages, could we get a Drawers manager here?

**Comment by hansyulian** [comment]
voting up! i also would like to see a drawer manager available.. since in my responsive use case, the web version use modal and the mobile version use drawer, i would like to be able to easily invoke some notification/confirmation by doing something like:

```
const confirmation = useConfirmation();

const function promptDelete = () => {
    confirmation.show("Are you sure you want to delete this record?")
}
```

```
function useConfirmation(){
    const isPhoneOrLower = useMediaQuery('.....');

    const show = (message: string) => {
        if (isPhoneOrLower){
           drawer.open(....)
        } else {
            modals.open(....)
        }
    }

   return {
       show
   }
}
```

**Reply by AndrejNemec** [reply]
Try this library https://github.com/AndrejNemec/mantine-vaul
Example: https://mantine-vaul.vercel.app/
